azurerm_servicebus_subscription_rule correlation_filter with whitespace label causes crash

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.3.2

### AzureRM Provider Version

3.0.0

### Affected Resource(s)/Data Source(s)

azurerm_servicebus_subscription_rule

### Terraform Configuration Files

```hcl
resource "azurerm_servicebus_subscription_rule" "foobar-subscription-rules" {
  name            = "foobar-subscription-rules"
  subscription_id = azurerm_servicebus_subscription.foobar-subscription.id
  filter_type     = "CorrelationFilter"

  correlation_filter {
    label = ""
  }
}
```


### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.0.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 219 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/servicebus.expandAzureRmServiceBusCorrelationFilter(0x663c7c0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/internal/services/servicebus/servicebus_subscription_rule_resource.go:382 +0x854
github.com/hashicorp/terraform-provider-azurerm/internal/services/servicebus.resourceServiceBusSubscriptionRuleCreateUpdate(0xc00240bf00, {0x567d7c0, 0xc001433880})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/internal/services/servicebus/servicebus_subscription_rule_resource.go:286 +0x705
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc002429e60, {0x668d1b8, 0xc002271a80}, 0x2, {0x567d7c0, 0xc001433880})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:329 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000e376c0, {0x668d1b8, 0xc002271a80}, 0xc00240d2b0, 0xc00240bd80, {0x567d7c0, 0xc001433880})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:467 +0x871
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0007262b8, {0x668d1b8, 0xc002271a80}, 0xc002259180)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000212580, {0x668d260, 0xc002413410}, 0xc001327030)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:603 +0x30e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x5bc6ca0, 0xc000212580}, {0x668d260, 0xc002413410}, 0xc002408a80, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001476380, {0x66de168, 0xc000c2f380}, 0xc002276360, 0xc001328390, 0xad51f60, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0xc6f
google.golang.org/grpc.(*Server).handleStream(0xc001476380, {0x66de168, 0xc000c2f380}, 0xc002276360, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x294

Error: The terraform-provider-azurerm_v3.0.0_x5 plugin crashed!
```


### Expected Behaviour

I thought we would provision a filter with a blank white space filter, or if that was forbidden get some kind of warning

### Actual Behaviour

The script crashed

### Steps to Reproduce

Provision a subscription rule with a whitespace label in the correlation_filter

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/servicebus/servicebus_subscription_rule_resource.go
+++ b/internal/services/servicebus/servicebus_subscription_rule_resource.go
@@ -330,6 +330,9 @@ func expandAzureRmServiceBusCorrelationFilter(d *pluginsdk.ResourceData) (*rules
 		return nil, fmt.Errorf("`correlation_filter` is required when `filter_type` is set to `CorrelationFilter`")
 	}
 
+	if configs[0] == nil {
+		return nil, fmt.Errorf("at least one property must not be empty in the `correlation_filter` block")
+	}
 	config := configs[0].(map[string]interface{})
 
 	contentType := config["content_type"].(string)

--- a/internal/services/servicebus/servicebus_subscription_rule_resource_test.go
+++ b/internal/services/servicebus/servicebus_subscription_rule_resource_test.go
@@ -3,6 +3,7 @@ package servicebus_test
 import (
 	"context"
 	"fmt"
+	"regexp"
 	"testing"
 
 	"github.com/hashicorp/go-azure-sdk/resource-manager/servicebus/2021-06-01-preview/subscriptions"
@@ -152,6 +153,18 @@ func TestAccServiceBusSubscriptionRule_updateSqlFilterToCorrelationFilter(t *tes
 	})
 }
 
+func TestAccServiceBusSubscriptionRule_correlationFilterWhiteSpace(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_servicebus_subscription_rule", "test")
+	r := ServiceBusSubscriptionRuleResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config:      r.correlationFilterWhiteSpace(data),
+			ExpectError: regexp.MustCompile("expanding `correlation_filter`: at least one property must not be empty in the `correlation_filter` block"),
+		},
+	})
+}
+
 func (t ServiceBusSubscriptionRuleResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
 	id, err := subscriptions.ParseRuleID(state.ID)
 	if err != nil {
@@ -296,6 +309,23 @@ resource "azurerm_servicebus_subscription_rule" "test" {
   correlation_filter {
     correlation_id = "test_correlation_id"
     message_id     = "test_message_id"
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r ServiceBusSubscriptionRuleResource) correlationFilterWhiteSpace(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_servicebus_subscription_rule" "test" {
+  name            = "acctestservicebusrule-%d"
+  subscription_id = azurerm_servicebus_subscription.test.id
+  action          = "SET Test='true'"
+  filter_type     = "CorrelationFilter"
+
+  correlation_filter {
+    label = ""
   }
 }
 `, r.template(data), data.RandomInteger)

